django/conf/global_settings.py

```diff
@@ -305,6 +305,11 @@
 # file system instead of into memory.
 FILE_UPLOAD_MAX_MEMORY_SIZE = 2621440  # i.e. 2.5 MB
 
+# Maximum size in bytes of request data before a SuspiciousOperation will
+# be raised.  This applies to accessing request.body and also to any
+# data in form data requests, excluding file uploads.
+DATA_UPLOAD_MAX_MEMORY_SIZE = 2621440  # i.e. 2.5 MB
```

Comment:
Servers DO limit the size of the URL including GET parameters.

Maybe this protection is not enough, but the "problems" you mention seem intrinsic to the Python VM and not something we can fix easily here. So, having this protection in place is much better than not having any.

As for making this limit per-view, that seems like a nice-to-have feature. We accept patches. :)